IdentityCore/src/validation/MSIDB2CAuthority.m

```diff
@@ -119,16 +127,24 @@ - (BOOL)supportsClientIDAsScope
 #pragma mark - Private
 
 + (NSURL *)normalizedAuthorityUrl:(NSURL *)url
+                  formatValidated:(BOOL)formatValidated
                           context:(id<MSIDRequestContext>)context
                             error:(NSError **)error
 {
-    if (![self isAuthorityFormatValid:url context:context error:error])
+    // remove query and fragments
+    if (!formatValidated)
     {
-        return nil;
+        NSURL *urlMinusFragment = [NSURL URLWithString:[url.absoluteString componentsSeparatedByString:@"#"][0]];
```

Comment:
nit: please use NSURLComponents instead. Also, there might be no path components at all, so this is a bit unsafe.

Reply:
Done